course/src/main/java/in/testpress/course/adapter/BaseContentListAdapter.kt

```diff
@@ -0,0 +1,100 @@
+package `in`.testpress.course.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.paging.PagingDataAdapter
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import `in`.testpress.core.TestpressSdk
+import `in`.testpress.course.R
+import `in`.testpress.course.databinding.RunningUpcomingListItemBinding
+import `in`.testpress.course.domain.DomainContent
+import `in`.testpress.course.domain.asDomainContent
+import `in`.testpress.course.ui.ContentActivity
+import `in`.testpress.course.util.DateUtils
+import `in`.testpress.util.ViewUtils
+
+class BaseContentListAdapter<T : Any>(COMPARATOR: DiffUtil.ItemCallback<T>):
+    PagingDataAdapter<T, BaseContentListViewHolder>(COMPARATOR){
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseContentListViewHolder {
+        val binding = RunningUpcomingListItemBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            false
+        )
+        return RunningContentListViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: BaseContentListViewHolder, position: Int) {
+        val content = getItem(position)
+        if (content != null) {
+            val domainContent = content.asDomainContent()
+            holder.bind(domainContent)
+        }
+    }
+}
+
+abstract class BaseContentListViewHolder(binding: RunningUpcomingListItemBinding) :
+    RecyclerView.ViewHolder(binding.root) {
+    private val title = binding.title
+    private val path = binding.path
+    private val date = binding.date
+    private val thumbnail = binding.thumbnail
+
+    init {
+        title.typeface = TestpressSdk.getRubikMediumFont(binding.root.context)
+        path.typeface = TestpressSdk.getRubikRegularFont(binding.root.context)
+        date.typeface = TestpressSdk.getRubikMediumFont(binding.root.context)
+    }
+
+    fun bind(content: DomainContent) {
+        title.text = content.title
+        path.text = "${content.treePath}"
+        thumbnail.setImageResource(getContentImage(content.contentType))
+        setDate(content)
+        setDateVisiblity(content)
```

Comment:
Let these derived view holder override bind and display these. By default date should not be visible and if needed then viewholder will derive and display it.


Don't use `set` in method name in derived class